[documentation] Secrets management in Operator 1.11+ (#1555)
DataDog/datadog-operator: docs/secret_management.md

**Note:** This secret helper requires Datadog Operator v0.5.0+

### How to deploy the agent components using the secret backend feature with DatadogAgent (Operator 1.11+)

> If using a custom script, create a Datadog Agent (or Cluster Agent) image following the example for the Datadog Operator above and reference credentials with `ENC[<placeholder>]`.

Reviewer: ```suggestion
If using a custom script, create a Datadog Agent (or Cluster Agent) image following the example for the Datadog Operator above, and specify credentials using `ENC[<placeholder>]`.
```

  # ...
```

The secret backend command can be specified in the `spec.global.secretBackend.command`:

```yaml
apiVersion: datadoghq.com/v2alpha1
kind: DatadogAgent
metadata:
  name: datadog
spec:
  global:
    secretBackend:
      command: "/my-secret-backend.sh"
  # ...
```

> The environment variable `DD_SECRET_BACKEND_COMMAND` from this configuration will be automatically applied to all the deployed components: node Agent, cluster Agent, cluster checks runners. Ensure the image you are using for all the components includes your command.

Reviewer: ```suggestion
The environment variable `DD_SECRET_BACKEND_COMMAND` from this configuration is automatically applied to all deployed components: node Agent, cluster Agent, and cluster checks runners. Ensure the image you are using for all the components includes your command.
```

> For convenience, the Datadog Agent and its sibling Cluster images already include a [helper function][2] `readsecret_multiple_providers.sh` that can be used to read from both files as well as Kubernetes Secrets. After creating the secret, set `spec.global.secretBackend.command` to `"/readsecret_multiple_providers.sh"`.

Reviewer: ```suggestion
For convenience, the Datadog Agent and its sibling Cluster images include a `readsecret_multiple_providers.sh` [helper function][2] that can be used to read from both files, as well as Kubernetes Secrets. After creating the Secret, set `spec.global.secretBackend.command` to `"/readsecret_multiple_providers.sh"`.
```

  name: datadog
spec:
  global:
    secretBackend:
      command: "/readsecret_multiple_providers.sh"
    credentials:
      apiKey: ENC[k8s_secret@default/test-secret/api_key]
      appKey: ENC[k8s_secret@default/test-secret/app_key]
```

**Remarks:**

* The helper `"/readsecret_multiple_providers.sh"` allows the (cluster) Agent to directly read Kubernetes secrets within both its own and other namespaces. Note that to do this, the associated ServiceAccount must be granted permissions with the appropriate Roles and RoleBindings. This can be done manually or using the following options:

>     * `global.secretBackend.enableGlobalPermissions`: whether a ClusterRole allowing the Agents to read **ALL** Kubernetes secrets is created.

Reviewer: ```suggestion
    * `global.secretBackend.enableGlobalPermissions`: Determines if a ClusterRole is created that enables the Agents to read **all** Kubernetes Secrets.
```

      ```yaml
      apiVersion: datadoghq.com/v2alpha1
      kind: DatadogAgent
      metadata:
        name: datadog
      spec:
        global:
          secretBackend:
            command: "/readsecret_multiple_providers.sh"
            enableGlobalPermissions: true
      # ...
      ```
    * `global.secretBackend.roles`: Replaces `enableGlobalPermissions`, detailing the list of namespaces/secrets to which the Agents should have access.
      ```yaml
      apiVersion: datadoghq.com/v2alpha1
      kind: DatadogAgent
      metadata:
        name: datadog
      spec:
        global:
          secretBackend:
            command: "/readsecret_multiple_providers.sh"
            roles:
            - namespace: rabbitmq-system
              secrets:
              - "rabbitmqcluster-sample-default-user"
      # ...
      ```

>       In this example, a Role is created granting read access to the secret `rabbitmqcluster-sample-default-user` in the `rabbitmq-system` namespace.

Reviewer 2: I'm being nitpicky here, but I notice a couple places we capitalize "secret" as a proper noun (as in the kubernetes docs https://kubernetes.io/docs/concepts/configuration/secret/) and some places we don't..

Author: Adressed in 20a4b0993fd214ecf5c54bd9fa418b623f7d88ef. Only left `namespace/secrets` as they reference the CRD keys

      **Note**: Each namespace in the list must be included in the DatadogAgent controller by setting `WATCH_NAMESPACE` or `DD_AGENT_WATCH_NAMESPACE` environment variables on the **Datadog Operator** container.

> * For the Agent and Cluster Agent, others options exist to configure the secret backend command:
>   * `global.secretBackend.args`: These arguments are supplied to the command when the Agent executes the secret backend command.
>   * `global.secretBackend.timeout`: secret backend execution timeout in second. The default value is 30 seconds.
> * For versions prior to Operator 1.11+, `spec.global.secretBackend` is unavailable. You should follow [these instructions][3] instead.

Reviewer 2: ```suggestion
* For the Agent and Cluster Agent, there are other configuration options for the secret backend command:
  * `global.secretBackend.args`: these arguments are supplied to the command when the Agent executes the secret backend command.
  * `global.secretBackend.timeout`: secret backend execution timeout in seconds. The default value is 30 seconds.
* For versions prior to Operator 1.11, `spec.global.secretBackend` is unavailable. You should follow [these instructions][3] instead.
```

Did you consider keeping the <1.11 instructions in this doc as a separate section?

Author: > Did you consider keeping the <1.11 instructions in this doc as a separate section?

Yes but felt it would have been this guide even longer than it is currently, so preferred to link instead of a section
